File: vbox-app/src/com/kedzie/vbox/api/IFile.java
package com.kedzie.vbox.api;

import com.kedzie.vbox.api.jaxb.FileSeekType;
import com.kedzie.vbox.soap.KSOAP;

@KSOAP(prefix="IFile")
public interface IFile extends IManagedObjectRef {

	@KSOAP(cacheable=true) public Integer getCreationMode();

	@KSOAP(cacheable=true) public Integer getDisposition();

	@KSOAP(cacheable=true) public String getFileName();

	@KSOAP(cacheable=true) public Long getInitialSize();
    
	@KSOAP(cacheable=true) public Integer getOpenMode();

	@KSOAP(cacheable=true) public Long getOffset();

	public void close();

	@KSOAP(cacheable=true) public IFsObjInfo getQueryInfo();

	public String read(@KSOAP(type="unsignedInt", value="toRead") int toRead, @KSOAP(type="unsignedInt", value="timeoutMS") int timeoutMS);

	public String readAt(@KSOAP(type="long", value="offset") long offset, @KSOAP(type="unsignedInt", value="toRead") int toRead, @KSOAP(type="unsignedInt", value="timeoutMS") int timeoutMS);

	public void seek(@KSOAP(type="unsignedInt", value="offset") long offset, FileSeekType whence);
    
	public void setACL(String acl);
	
	public Integer write(String data, @KSOAP(type="unsignedInt", value="timeoutMS") int timeoutMS);
    
	public Integer writeAt(@KSOAP(type="unsignedInt", value="offset") long offset, String data, @KSOAP(type="unsignedInt", value="timeoutMS") int timeoutMS);
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IFile.kt
package com.kedzie.vbox.api

import android.os.Parcelable
import com.kedzie.vbox.api.jaxb.FileSeekOrigin
import com.kedzie.vbox.soap.Cacheable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy

@KsoapProxy
@Ksoap(prefix = "IFile")
interface IFile : IManagedObjectRef {
    @Cacheable( "CreationMode")
	suspend fun getCreationMode(): Int

    @Cacheable("Disposition")
	suspend fun getDisposition(): Int

    @Cacheable( "FileName")
	suspend fun getFileName(): String

    @Cacheable("InitialSize")
	suspend fun getInitialSize(): Long

    @Cacheable("OpenMode")
	suspend fun getOpenMode(): Int

    @Cacheable("Offset")
	suspend fun getOffset(): Long

    suspend fun close()

    @Cacheable("QueryInfo")
	suspend fun getQueryInfo(): IFsObjInfo

    suspend fun read(@Ksoap(type="unsignedInt") toRead: Int, @Ksoap(type="unsignedInt") timeoutMS: Int): String

    suspend fun readAt(@Ksoap(type="long") offset: Long, @Ksoap(type="unsignedInt") toRead: Int, @Ksoap(type="unsignedInt") timeoutMS: Int): String

    suspend fun seek(@Ksoap(type="unsignedInt") offset: Long, whence: FileSeekOrigin)

    suspend fun setACL(acl: String)

    suspend fun write(data: String, @Ksoap(type="unsignedInt") timeoutMS: Int): Int

    suspend fun writeAt(@Ksoap(type="unsignedInt") offset: Long, data: String, @Ksoap(type="unsignedInt") timeoutMS: Int): Int
}